SCClassLibrary/Common/Quarks/Quark.sc

```diff
@@ -89,16 +96,20 @@ Quark {
 		if(Git.isGit(localPath), {
 			data = git = refspec = nil;
 			changed = true;
+			this.runHook(\preUpdate);
 			git = Git(localPath);
 			git.pull();
 			git.checkout("master");
+			this.runHook(\postUpdate);
```

Comment:
This COULD be a problem, sadly... `git.checkout` here can change the Quark data, and this potentially change the hooks. It MIGHT be that what we actually want to do here is:
1. Run `\preUpdate`
2. `git.pull()` / `git.checkout`
3. Reload `data`
4. run `\postUpdate`

It would probably have to be clearly documented that `\preUpdate` is always the OLD hook, and `\postUpdate` is always the NEW hook.

Reply:
@capital-G did you see scotts remark?